src/node/commit.rs

```diff
@@ -538,4 +538,18 @@ mod tests {
         // I created this unit test by hand
         assert_program_not_deserializable::<Core>(&[0xa9, 0x48, 0x00], &Error::NotInCanonicalOrder);
     }
+
+    #[test]
+    fn regression_177_1() {
```

Comment:
50293ca9872e6ad6dddd42aa020f17a7d43336e6: Adding this test case _before_ the infinite loop is fixed means that CI runs forever on this commit. @apoelstra Does your local CI have a timeout?

Reply:
Oh, lol, let me reorder the commits. I don't know if I have a timeout, but if I do, it is in exceess of an hour..